### PR TITLE
Improve the keyword in subheadings assessment

### LINKED ISSUE #1311: Improve keyword in subheadings assessment
### Description

Currently, we only check whether the keyword is included in one or more subheadings. We also want to check whether it is included in too few or too many subheadings with regards to the total number of subheadings in the text.

### Boundaries

- `subheadingWithKeyword = 0`: red

- `subheadingWithKeyword < 30% of totalNumberOfSubheadings`: red

-  `30% of totalNumberOfSubheadings < subheadingWithKeyword < 75% of totalNumberOfSubheadings` or if there is only 1 subheading and that subheading contains the keyword: green

- `subheadingWithKeyword > 75% of totalNumberOfSubheadings`: red (only check if totalNumberOfSubheadings > 1)

### Feedback

- `subheadingWithKeyword = 0`: You have not used the focus keyword in any subheading (such as an H2) in your copy. (unchanged)

- `subheadingWithKeyword < 30% of totalNumberOfSubheadings`: The focus keyword appears only in %1$d (out of %2$d) subheadings in your copy. Try to use it in more subheadings.

- `30% of totalNumberOfSubheadings < subheadingWithKeyword < 75% of totalNumberOfSubheadings`: The focus keyword appears  in %1$d (out of %2$d) subheadings in your copy. That's great.

- `subheadingWithKeyword > 75% of totalNumberOfSubheadings`: The focus keyword appears in  %1$d (out of %2$d) subheadings in your copy. That might sound a bit repetitive. Try to change some of those subheadings to make your text sound more natural.

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry:
* Improves the keyword in subheadings assessment so that it checks whether the keyword is included in too few or too many subheadings with regards to the total number of subheadings in the text.

## Relevant technical choices
* Adds missing translator comments to `metaDescriptionKeywordAssessment`

## Test instructions

This PR can be tested by following these steps:

* Check out the branch.
* Use the browserified example. Don't forget to run a `grunt build:js`.
* Add a text and a keyword.
* Include some subheadings in your text. Try various combinations with subheadings containing the keyword and subheadings not containing the keyword. Make sure the plugin provides meaningful feedback strings.

Note: This assessment is currently still triggered when using multiple keywords with Premium. However, this won't be the case anymore once a secondary keyword assessor has been implemented (https://github.com/Yoast/YoastSEO.js/issues/1479) in order to prevent keyword stuffing.

Fixes #1311

### CODE REVIEW

File: spec/assessments/subheadingsKeywordSpec.js [hunk 1]
```diff
@@ -5,34 +5,54 @@ var i18n = Factory.buildJed();
 
 let matchKeywordAssessment = new SubheadingsKeywordAssessment();
 
-describe( "An assessment for matching keywords in subheadings", function(){
-	it( "assesses a string without subheadings", function(){
+describe( "An assessment for matching keywords in subheadings", function() {
+	it( "returns no score on a string without subheadings", function() {
```
Comment (IreneStr) [inline spec/assessments/subheadingsKeywordSpec.js]: It does return a score of 0

File: src/assessments/seo/subheadingsKeywordAssessment.js [hunk 2]
```diff
@@ -68,15 +72,33 @@ class SubHeadingsKeywordAssessment extends Assessment {
 	 * @returns {number|null} The calculated score.
 	 */
 	calculateScore( subHeadings ) {
+		let minNumberOfSubheadings = subHeadings.count * this._config.lowerBoundary;
+		let maxNumberOfSubheadings = subHeadings.count * this._config.upperBoundary;
```
Comment (IreneStr) [inline src/assessments/seo/subheadingsKeywordAssessment.js]: Question: do we want these to be `Math.floor`ed or `Math.ceil`ed?

File: src/assessments/seo/subheadingsKeywordAssessment.js [hunk 3]
```diff
@@ -92,17 +114,49 @@ class SubHeadingsKeywordAssessment extends Assessment {
 	 * @returns {string} The translated string.
 	 */
 	translateScore( score, subHeadings, i18n ) {
-		if ( score === this._config.scores.multipleMatches || score === this._config.scores.oneMatch ) {
+		let minNumberOfSubheadings = subHeadings.count * this._config.lowerBoundary;
```
Comment (IreneStr) [inline src/assessments/seo/subheadingsKeywordAssessment.js]: These calculations are also done inside the calculateScore function. Can you make minNumberOfSubheadings and maxNumberOfSubheadings globally available within this file?

File: src/assessments/seo/subheadingsKeywordAssessment.js [hunk 4]
```diff
@@ -92,17 +114,49 @@ class SubHeadingsKeywordAssessment extends Assessment {
...
+		let maxNumberOfSubheadings = subHeadings.count * this._config.upperBoundary;
+
+		if ( subHeadings.matches === 0 ) {
```
Comment (IreneStr) [inline src/assessments/seo/subheadingsKeywordAssessment.js]: You're not using the scores you've set above. You're repeating all if-statements instead. Why?

**Comment by Dieterrr** [comment]
Acceptance done! 👍

**Pull request merged**